variation-commons-mongodb/src/main/java/uk/ac/ebi/eva/commons/mongodb/writers/VariantMongoWriter.java

```diff
@@ -129,6 +132,31 @@ private DBObject generateUpdate(IVariant variant) {
         Assert.notNull(variant, "Variant should not be null. Please provide a valid Variant object");
         logger.trace("Convert variant {} into mongo object", variant);
 
+        BasicDBObject update = new BasicDBObject();
+
+        addOperatorAddToSet(variant, update);
+        addOperatorSetOnInsert(variant, update);
+        addOperatorSet(variant, update);
+
+        return update;
+    }
+
+    private void addOperatorSetOnInsert(IVariant variant, BasicDBObject update) {
+        update.append("$setOnInsert", convertVariant(variant));
+    }
+
+    private void addOperatorSet(IVariant variant, BasicDBObject update) {
+        BasicDBObject overwrite = new BasicDBObject();
+        if (variant.getMainId() != null) {
+            overwrite.put(MAIN_ID_FIELD, variant.getMainId());
+        }
+
+        if (!overwrite.isEmpty()) {
+            update.append("$set", overwrite);
```

Comment:
Could this append be moved into the previous "if" enclosure? that is the only place where "overwrite" seems to be set.

Reply:
well, it could. this function follows the same structure of addOperatorAddToSet, where there are several things added. This way was faster to do, works the same and adding things will cost less.

Reply:
i see. got it.